k2/python/k2/symbol_table.py

```diff
@@ -104,7 +168,39 @@ def get(self, k: Union[int, str]) -> Union[str, int]:
         '''
         if isinstance(k, int):
             return self._id2sym[k]
-        elif isinstance(k, str):
-            return self._sym2id[k]
         else:
-            raise ValueError(f'Unsupported type {type(k)}.')
+            return self._sym2id[k]
+
+    def merge(self, other: 'SymbolTable') -> 'SymbolTable':
+        '''Create a union of two SymbolTables.
+        Raises an AssertionError if the same IDs are occupied by
+         different symbols.
+
+        Args:
+            other:
+                A symbol table to merge with ``self``.
+
+        Returns:
+            A new symbol table.
+        '''
+        common_ids = set(self._id2sym).intersection(other._id2sym)
+        assert self.eps == other.eps, f'Mismatched epsilon symbol: ' \
+                                      f'{self.eps} != {other.eps}'
+        for idx in common_ids:
+            assert self[idx] == other[idx], f'ID conflict for id: {idx}, ' \
```

Comment:
This only checks that the same ID maps to the same symbol.

Do we need to add an extra check that the same symbol maps to the same ID?